File: app/src/main/java/com/example/yuki/glidetorecyclerview/MainActivity.java
package com.example.yuki.glidetorecyclerview;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private Button toRecyclerLayoutBtn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        toRecyclerLayoutBtn = findViewById(R.id.toRecyclerLayoutBtn);
        toRecyclerLayoutBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //MainActivity.thisと書くと強参照のためOutOfMemoryが発生する可能性があるのでgetApplication()にする
                Intent intent = new Intent(getApplication(), RecyclerViewLayout.class);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/example/yuki/glidetorecyclerview/MainActivity.kt
package com.example.yuki.glidetorecyclerview

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val toRecyclerLayoutBtn = findViewById<Button>(R.id.toRecyclerLayoutBtn)
        toRecyclerLayoutBtn!!.setOnClickListener {
            //MainActivity.thisと書くと強参照のためOutOfMemoryが発生する可能性があるのでApplicationにする
            val intent = Intent(application, RecyclerViewLayout::class.java)
            startActivity(intent)
        }
    }
}
